docs/THEORY.md

```
@@ -0,0 +1,978 @@
+2. Given a skeleton S and recursively computed results for all attachments, how to combine them
+
+The result is then **unique**. This gives us structural recursion and induction "for free."
+
+### Depth and Unfoldings
+
+**Depth** of a WARP X:
+- Atoms have depth 0
+- A composite WARP `(S, α, β)` has depth = 1 + max depth of all attachments
+
+**k-unfolding** `unf_k(X)`:
+- Keep all structure at depths 0, ..., k-1 unchanged
```

markdownlint-cli2 (0.18.1)

Trailing spaces
Expected: 0 or 2; Actual: 1

(MD009, no-trailing-spaces)